kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/KubernetesListOperationsImpl.java

```diff
@@ -79,17 +80,22 @@ public KubernetesList create(KubernetesList... items) {
       items = new KubernetesList[]{get()};
     }
     for (KubernetesList i : items) {
-      for (HasMetadata r : i.getItems()) {
-        HasMetadata created = create(r);
-        createdItems.add(created);
-      }
+      createdItems.addAll(createItemsInKubernetesList(i));
     }
 
     KubernetesList createdList = new KubernetesList();
     createdList.setItems(createdItems);
     return createdList;
   }
 
+  @Override
+  public KubernetesList create(KubernetesList list) {
+    if (list.getItems().isEmpty()) {
+      list = get();
+    }
+    return new KubernetesListBuilder().withItems(createItemsInKubernetesList(list)).build();
+  }
```

Comment:
You can reuse the #create(KubernetesList... items) method so logic will be the same in both scenarios:
```suggestion
  @SuppressWarnings("squid:S3878")
  @Override
  public KubernetesList create(KubernetesList list) {
    return create(new KubernetesList[]{list});
  }
```